pkg/reconciler/revision/resolve_test.go

```diff
@@ -311,11 +332,57 @@ func TestResolveNoAccess(t *testing.T) {
 		ServiceAccountName: svcacct,
 	}
 	// If there is a failure accessing the ServiceAccount for this Pod, then we should see an error.
-	if resolvedDigest, err := dr.Resolve("ubuntu:latest", opt, emptyRegistrySet); err == nil {
+	if resolvedDigest, err := dr.Resolve(context.Background(), "ubuntu:latest", opt, emptyRegistrySet); err == nil {
 		t.Fatalf("Resolve() = %v, want error", resolvedDigest)
 	}
 }
 
+func TestResolveTimeout(t *testing.T) {
+	// Stand up a fake registry which blocks until cancelled.
+	server, cancel := fakeRegistryBlocking(t)
+	defer cancel()
+
+	u, err := url.Parse(server.URL)
+	if err != nil {
+		t.Fatalf("url.Parse(%v) = %v", server.URL, err)
+	}
+
+	// Create a tag pointing to an image on our fake registry
+	tag, err := name.NewTag(fmt.Sprintf("%s/%s:latest", u.Host, "doesnt/matter"), name.WeakValidation)
+	if err != nil {
+		t.Fatalf("NewTag() = %v", err)
+	}
+
+	// Set up a fake service account with pull secrets for our fake registry
+	ns, svcacct := "user-project", "user-robot"
+	client := fakeclient.NewSimpleClientset(&corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      svcacct,
+			Namespace: ns,
+		},
+	})
+
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
```

Comment:
I think we can significantly tighten this. Maybe 100ms instead?

Reply:
I want to make sure the thing we cancel inside is the actual manifest get, if this is too short we can accidentally end up only testing cancelling the ping :(

Reply:
We *could* make this super safe by adding a second channel that is closed by the server so you know in the test when the actual request has been received. That'd allow us to ditch the hardcoded timeout completely and just use a cancel function on the context after that signal has arrived.

Up to you if you think that's worth here.

Reply:
yeah, considered something like that but seems like a tiny bit more ceremony than this is worth here. I've bumped the timeout way down to 200ms tho, that seems good enough to be very confident we're timing out in go-containerregistry rather than credential lookup.